### ISSUE TITLE
Cannot import 'jose' in 3.9.6

### ISSUE THREAD

**Issue by youngbob** [patriksimek/vm2#397]
Something about the import handling changes in 3.9.6 broke the ability to import https://github.com/panva/jose.

```javascript
const jose = require('jose')
```

fails with: `VMError: Package path '.jose' is not exported`

**Comment by XmiliaH** [comment]
Thanks for the report.
The require mechanism was improved to follow new features of Node's version. This includes the usage of the exports field in the `package.json`. It seems that I did an offset error in `resolver.js`. The fixed version should be:
```js
// LOAD_PACKAGE_EXPORTS(X, DIR)
loadPackageExports(x, dir, extList) {
	// 1. Try to interpret X as a combination of NAME and SUBPATH where the name
	//    may have a @scope/ prefix and the subpath begins with a slash (`/`).
	const res = x.match(EXPORTS_PATTERN);
	// 2. If X does not match this pattern or DIR/NAME/package.json is not a file,
	//    return.
	if (!res) return undefined;
	const scope = this.pathConcat(dir, res[1]);
	const pack = this.readPackage(scope);
	if (!pack) return undefined;
	// 3. Parse DIR/NAME/package.json, and look for "exports" field.
	// 4. If "exports" is null or undefined, return.
	if (!pack.exports) return undefined;
	// 5. let MATCH = PACKAGE_EXPORTS_RESOLVE(pathToFileURL(DIR/NAME), "." + SUBPATH,
	//    `package.json` "exports", ["node", "require"]) defined in the ESM resolver.
	const match = this.packageExportsResolve(scope, '.' + (res[2] || ''), pack.exports, ['node', 'require'], extList);
	// 6. RESOLVE_ESM_MATCH(MATCH)
	return this.resolveEsmMatch(match, x, extList);
}
```
I will add some tests and push it tomorrow.

**Comment by XmiliaH** [comment]
Should be fixed in v3.9.7
